### DISCUSSION TITLE
[E002] Can't find factory for 'tok2vec'. Where can I get tok2vec?

### DISCUSSION THREAD

**Discussion by psydok** [explosion/spaCy#8961] [Help: Installation] [answered]
## How to reproduce the behaviour
```
import spacy
nlp = spacy.load('/usr/local/lib/python3.7/dist-packages/ru_core_news_sm')
```

```
KeyError                                  Traceback (most recent call last)
<ipython-input-200-01b912c0ac04> in <module>()
      2 
      3 
----> 4 nlp = spacy.load('/usr/local/lib/python3.7/dist-packages/ru_core_news_sm')

3 frames
/usr/local/lib/python3.7/dist-packages/spacy/language.py in create_pipe(self, name, config)
    298 
    299     @property
--> 300     def component_names(self) -> List[str]:
    301         """Get the names of the available pipeline components. Includes all
    302         active and inactive pipeline components.

KeyError: "[E002] Can't find factory for 'tok2vec'. This usually happens when spaCy calls `nlp.create_pipe` with a component name that's not built in - for example, when constructing the pipeline from a model's meta.json. If you're using a custom component, you can write to `Language.factories['tok2vec']` or remove it from the model meta and add it via `nlp.add_pipe` instead."
```

I don't understand what needs to be done. Where can I get tok2vec? I don't understand what needs to be done.

```
import ru_core_news_lg
import spacy
nlp = ru_core_news_lg.load()
```
I tried it like this. The error is the same

## Your Environment
* Operating System: Lunix (Colab)
* Python Version Used: 3.7
* spaCy Version Used: 3.0.0
* Environment Information:
## Info about spaCy

- **spaCy version:** 3.0.0
- **Platform:** Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.7.11
- **Pipelines:** ru_core_news_lg (3.0.0), ru_core_news_sm (3.0.0)

**Comment by polm** [comment, marked as answer]
This is a weird error and should not be happening if spaCy is installed properly. Can you try reinstalling spaCy? 

Also if you're just getting started definitely upgrade to the latest version, v3.1.1. v3.0.0 was the initial major release and there have been several bugfixes since then. 

You can upgrade spaCy like this:

    pip install -U spacy

Note that if you upgrade to v3.1 or higher you will have to download the models again, as they are retrained every time the minor version number goes up.

**Reply by psydok** [reply]
Didn't work like that yesterday. Apparently, I did not delete something from past installations.
It worked now.
Thank you!

```
!pip install -U spacy
!python -m spacy download ru_core_news_sm
import spacy
nlp = spacy.load("ru_core_news_sm")
doc = nlp("добрый вечер иван иванович")

for ent in doc.ents:
    print(ent.text, ent.start_char, ent.end_char, ent.label_)
```

**Reply by polm** [reply]
Glad you got it working, thanks for letting us know that resolved it! Feel free to post again if you have any other issues or questions.
